File: Level.3/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool is_intersect(vector<int> &a, vector<int> &b) {
    return a[1] >= b[0];
}

int solution(vector<vector<int>> routes) {
    int answer = 1;
    sort(routes.begin(), routes.end());
    for(int i = 0; i < routes.size() - 1; ++i) {
        if(is_intersect(routes[i], routes[i+1])) {
            if(routes[i+1][1] > routes[i][1]) {
                routes[i+1][1] = routes[i][1];   
            }
        }
        else {
            ++answer;
        }
    }
    return answer;
}

File: JavaScript/Level.3/단속카메라.js
function solution(routes) {
    let answer = 0;
    routes.sort((a, b) => {
        return a[1] - b[1];
    });
    let camTime = -30001;
    routes.forEach((route) => {
        if (camTime < route[0]) {
            camTime = route[1];
            ++answer;
        }
    });
    return answer;
}
